pygen/pygen_src/isa/riscv_instr_cov.py

```diff
@@ -0,0 +1,44 @@
+import vsc
+import logging
+from enum import Enum, auto
+
+class operand_sign_e(Enum):
+    POSITIVE = auto()
+    NEGATIVE = auto()
+
+class div_result_e(Enum):
+    DIV_NORMAL = auto()
+    DIV_BY_ZERO = auto()
+    DIV_OVERFLOW = auto()
+
+class compare_result_e(Enum):
+    EQUAL = auto()
+    LARGER = auto()
+    SMALLER = auto()
+
+class logical_similarity_e(Enum):
+    IDENTICAL = auto()
+    OPPOSITE = auto()
+    SIMILAR = auto()
+    DIFFERENT = auto()
+
+class special_val_e(Enum):
+    NORMAL_VAL = auto()
+    MIN_VAL = auto()
+    MAX_VAL = auto()
+    ZERO_VAL = auto()
+
+# dictionary to keep track of reg_name:reg_value throughout the program 
+gpr_state = {}
+
+def get_gpr(reg_name):
+    reg_name = reg_name.upper()
+    # logging.error...
```

Comment:
maybe a helper function is needed here to convert a string into the corresponding `Enum` member (in this case converting `reg_name` into `riscv_reg_t` Enum type)?
https://docs.python.org/3/library/enum.html#programmatic-access-to-enumeration-members-and-their-attributes

Reply:
Thanks for the comment! Fixed.